device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
-receiver.on('response', function(response) {
-    console.log('Response received for request ' + response.requestId);
-    console.log('  status = " + response.status);
-    console.log(response.body);
-});
+## Public API
+
+```typescript
+class AmqpTwinClient extends EventEmitter {
+  constructor(authenticationProvider: AuthenticationProvider, client: any) {
+  getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
+  updateTwinReportedProperties(patch: any, callback: (err?: Error) => void): void;
+  enableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  disableTwinDesiredPropertiesUpdates(callback: (err?: Error) => void): void;
+  detach(callback: (err?: Error) => void): void;
+}
 ```
+### constructor(authenticationProvider: AuthenticationProvider, client: any) {
 
-## Public API
+**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
 
-### Constructor
+### getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_004: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `config` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_007: [** The `getTwin` method shall attach the sender link if it's not already attached. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_002: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `client` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_008: [** If attaching the sender link fails, the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_009: [** THe `getTwin` method shall attach the receiver link if it's not already attached. **]**
 
-### response event
+**SRS_NODE_DEVICE_AMQP_TWIN_16_010: [** If attaching the receiver link fails, the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_006: [** When a listener is added for the `response` event, and the `post` event is NOT already subscribed, upstream and downstream links are established via calls to `attachReceiverLink` and `attachSenderLink`. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_011: [** The `getTwin` method shall send an `AmqpMessage` using the `SenderLink.send` method with the following annotations and properties:
+- `operation` annotation set to `GET`.
+- `resource` annotation set to `undefined`
+- `correlationId` property set to a random integer
+- `body` set to ` `. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_007: [** The endpoint argument for attacheReceiverLink shall be `/devices/<deviceId>/twin`. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_012: [** If the `SenderLink.send` call fails the `getTwin` method shall call its callback with the error that caused the failure. **]**
```

Comment:
>If the `SenderLink.send` call fails the `getTwin` method [](start = 40, length = 56)

nit: add comma after the word "fails".  Otherwise, it first reads as the getTwin method being the object of the failure, not the subject of the following clause.